engine/src/main/java/org/terasology/engine/physics/bullet/BulletPhysics.java

```diff
@@ -743,16 +744,16 @@ private btCollisionShape getShapeFor(EntityRef entityRef) {
         }
         HullShapeComponent hull = entityRef.getComponent(HullShapeComponent.class);
         if (hull != null) {
-            FloatBuffer buffer = BufferUtils.createFloatBuffer(hull.sourceMesh.getVertices().size());
-            TFloatIterator iterator = hull.sourceMesh.getVertices().iterator();
-            int numPoints = 0;
-            while (iterator.hasNext()) {
-                numPoints++;
-                buffer.put(iterator.next());
-                buffer.put(iterator.next());
-                buffer.put(iterator.next());
+            VertexAttributeBinding<Vector3fc, Vector3f> positions = hull.sourceMesh.getVertices();
+            FloatBuffer buffer = BufferUtils.createFloatBuffer(positions.getResource().elements() * 3);
```

Comment:
Should this use the abstraction for `getVertexCount()` rather than looking at the resource's elements directly? As this number is used later again, maybe even save it to a local field?

```suggestion
            int vertexCount = hull.sourceMesh.getVertexCount();
            FloatBuffer buffer = BufferUtils.createFloatBuffer(vertexCount * 3);
```